app/policies/repository_policy.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# Policies for RepositoriesController
+class RepositoryPolicy < ApplicationPolicy
+  attr_reader :current_user, :repository
+
+  def initialize(current_user, repository)
+    @current_user = current_user
+    @repository = repository
+    super
+  end
+
+  def show?
+    repository.public_access ||
+      !!current_user&.accessible_repositories&.map(&:to_param)&.
+        include?(repository.to_param)
+  end
+
+  def create?(owner)
+    return false unless current_user
+    if owner.is_a?(User)
+      owner&.id == current_user&.id
```

Comment:
I think you don't need the `&.` operators here, as you already made sure that `owner` is a user (so it can't be `nil`) and `current_user` is set